lib/src/winrt/internal/vector_helper.dart

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// ignore_for_file: camel_case_types, non_constant_identifier_names
+
+import 'dart:ffi';
+
+import 'package:ffi/ffi.dart';
+
+import '../../combase.dart';
+import '../../extensions/comobject_pointer.dart';
+import '../../extensions/hstring_array.dart';
+import '../../extensions/int_array.dart';
+import '../../types.dart';
+import '../../utils.dart';
+import '../../winrt_helpers.dart';
+
+class VectorHelper<T> {
+  const VectorHelper(
+    this.creator,
+    this.enumCreator,
+    this.intType,
+    this.getManyCallback,
+    this.length, {
+    this.allocator = calloc,
+  });
+
+  final Allocator allocator;
+  final T Function(Pointer<COMObject>)? creator;
+  final T Function(int)? enumCreator;
+  final Type? intType;
+  final void Function(int, int, Pointer<NativeType>) getManyCallback;
+  final int length;
+
+  List<T> toList({bool unmodifiable = false}) {
+    final List<T> list;
+    if (isSameType<T, int>()) {
+      list = _toList_int() as List<T>;
+    } else if (isSameType<T, String>()) {
+      list = _toList_String() as List<T>;
+    } else if (isSubtypeOfWinRTEnum<T>()) {
+      list = _toList_enum();
+    } else {
+      list = _toList_COMObject();
+    }
+
+    return unmodifiable ? List.unmodifiable(list) : list;
+  }
+
+  List<String> _toList_String() {
+    final pArray = calloc<HSTRING>(length);
+
+    try {
+      getManyCallback(0, length, pArray);
+      return pArray.toList(length: length);
+    } finally {
+      free(pArray);
+    }
+  }
+
+  List<T> _toList_COMObject() {
+    final pArray = allocator<COMObject>(length);
+    getManyCallback(0, length, pArray);
+    return pArray.toList(creator!, length: length);
+  }
+
+  List<T> _toList_enum() {
+    switch (intType) {
```

Comment:
Maybe a comment here to highlight that for enums, the only valid WinRT types are `Int32` or `UInt32`, with a link to https://docs.microsoft.com/en-us/uwp/winrt-cref/winrt-type-system#enums

Reply:
Done